### PR TITLE
Feature/1322 register a hadoop datastore

### LINKED ISSUE #1322: DC monitor: Register a Hadoop datastore
Add registration for a file located to Hadoop cluster to this form. 

image

This might include: 
- [ ] Selection of the server
- [ ] Selection of the file

### PR DESCRIPTION
**Pull request by ClaudiaPHI**
Fixes #1322  

image

image

image

image

### CODE REVIEW

**Comment by ClaudiaPHI** [comment]
Does anyone want to review/test this PR ?

**Comment by LosD** [comment]
I'll take a look

File: monitor/extensions/csv-datastore-wizard/src/main/java/org/datacleaner/monitor/server/wizard/CsvDatastoreWizardSession.java [hunk 1]
```diff
@@ -86,26 +88,31 @@ protected WizardPageController nextPageControllerExisting() {
...
-                                    // only .csv and .tsv files are allowed to
-                                    // be referenced on the server, for security
-                                    // reasons.
-                                    throw new DCUserInputException(
-                                            "For security reasons, only existing .csv, .tsv or .txt files can be referenced on the server");
+                    protected WizardPageController nextPageController(Resource resource) {
+                        if (resource instanceof FileResource) {
+                            final File file = ((FileResource) resource).getFile();
+                            final String filepath = file.getAbsolutePath();
+                            if (!filepath.toLowerCase().endsWith(".csv")) {
```
Comment (LosD) [inline monitor/extensions/csv-datastore-wizard/src/main/java/org/datacleaner/monitor/server/wizard/CsvDatastoreWizardSession.java]: That's a pretty nasty construct. Can we use ORing instead (I'm also pretty convinced we have are tools for checking extensions, maybe they can take varargs or an array of extensions to check for)?

Why the special case for txt files?

Comment (ClaudiaPHI) [inline monitor/extensions/csv-datastore-wizard/src/main/java/org/datacleaner/monitor/server/wizard/CsvDatastoreWizardSession.java]: I will improve the with the "OR" and check about the extensions. Why the special cases ? I have no idea.

Comment (LosD) [inline monitor/extensions/csv-datastore-wizard/src/main/java/org/datacleaner/monitor/server/wizard/CsvDatastoreWizardSession.java]: Oh, didn't notice it in the old code.

**Comment by LosD** [comment]
Only very minor stuff in the code (and just now realised that it's not in new code, but in reformatted old code).

However, I'm pretty sure that that a .tsv is just as legal as a .csv file for Hadoop? It's the same format, just using tab and 0x10 for EOL while CSV is a little more free to use other characters.

**Pull request merged**